src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -521,38 +595,15 @@ protected void rescaleGlsk(List<SensitivityFactorGroup> factorGroups, GraphDecre
         }
     }
 
-    protected void computeInjectionFactors(Map<String, Double> participationMap, List<SensitivityFactorGroup> factorGroups) {
-        computeInjectionFactors(x -> participationMap, factorGroups);
-    }
-
-    protected LfBus getParticipatingElementLfBus(ParticipatingElement participatingElement) {
-        if (participatingElement.getElement() instanceof LfGenerator) {
-            return ((LfGenerator) participatingElement.getElement()).getBus();
-        } else if (participatingElement.getElement() instanceof LfBus) {
-            return (LfBus) participatingElement.getElement();
-        } else {
-            throw new UnsupportedOperationException("Unsupported participating element");
-        }
+    protected List<ParticipatingElement> getParticipatingElements(LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters, Function<ParticipatingElement, Boolean> filter) {
+        ActivePowerDistribution.Step step = ActivePowerDistribution.getStep(loadFlowParameters.getBalanceType());
+        List<ParticipatingElement> participatingElements =  step.getParticipatingElements(lfNetwork).stream().filter(filter::apply).collect(Collectors.toList());
+        ParticipatingElement.normalizeParticipationFactors(participatingElements, "bus");
+        return participatingElements;
     }
 
     protected List<ParticipatingElement> getParticipatingElements(LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters) {
-        ActivePowerDistribution.Step step;
-        List<ParticipatingElement> participatingElements;
-
-        switch (loadFlowParameters.getBalanceType()) {
-            case PROPORTIONAL_TO_GENERATION_P_MAX:
-                step = new GenerationActionPowerDistributionStep();
-                break;
-            case PROPORTIONAL_TO_LOAD:
-                step = new LoadActivePowerDistributionStep(false, false);
-                break;
-            default:
-                throw new UnsupportedOperationException("Balance type not yet supported: " + loadFlowParameters.getBalanceType());
-        }
-
-        participatingElements = step.getParticipatingElements(lfNetwork);
-        ParticipatingElement.normalizeParticipationFactors(participatingElements, "bus");
-        return participatingElements;
+        return getParticipatingElements(lfNetwork, loadFlowParameters, element -> true);
```

Comment:
I am not sure to get the aim of this function...

Reply:
It allows you to get the normalized participating elements

Reply:
Indeed I get it, it is to filter.

Reply:
It is possible to use the isDisabled attribute, but it less efficient. 
To use the isDisabled parameter, I think we would need to:
- Save all the buses states
- Go through all the buses and disconnect the one that needs to be disconnected
- Go through all the buses to get the non-disconnected participating elements
- Restore all the buses states

With the current filtering in the DC getParticipatingElements, we are just doing:
- Go through all the buses to get participating elements and get the one in the right connected component

Reply:
I was thinking about save the state of the bus that initial contains a participating element. Then, you disabled those that are not in the main connected component of this connectivity. Then, you can use the classical getParticipatingElements. Then you restore the buses that initially contain participating element. It is just to ensure coherence in the code, but it is not a bit deal.